api/src/main/java/org/openmrs/api/UserService.java

```diff
@@ -552,4 +554,6 @@
 	@Authorized
 	public void changePasswordUsingSecretAnswer(String secretAnswer, String pw) throws APIException;
 
+  void notifyUserSessionListener(User user, Event event, Status status);
```

Comment:
I'd think this being internal logic, this should be private

Reply:
And I'd expect it to be in the same class as the code that call it

Reply:
its rather tricky injecting beans into a non spring UserService managed class

Reply:
You call `Context.getRegisteredComponents(UserSessionListener.class)`